jetty-server/src/main/java/org/eclipse/jetty/server/HttpChannel.java

```diff
@@ -406,7 +410,13 @@ public boolean handle()
                             // the following is needed as you cannot trust the response code and reason
                             // as those could have been modified after calling sendError
                             Integer code = (Integer)_request.getAttribute(RequestDispatcher.ERROR_STATUS_CODE);
-                            _response.setStatus(code != null ? code : HttpStatus.INTERNAL_SERVER_ERROR_500);
+                            if (code == null)
+                                code = HttpStatus.INTERNAL_SERVER_ERROR_500;
+                            _response.setStatus(code);
+
+                            // Add Connection:close if we can't consume the input
+                            if (!_request.getHttpInput().consumeAll())
+                                ensureConnectionClose();
```

Comment:
Isn't this a duplication of what happens in `COMPLETE`?

Reply:
It is similar logic, but happens at a different time - ie between returning from a dispatch after a sendError and before an ERROR dispatch... which may do lots and lots of stuff before we ever get to COMPLETE.

Reply:
@gregw and so, if we may dispatch to an error handler that then reads the request content, we should not do it ourselves here, no?

Reply:
I don't think that is a valid use case for an error handler to do.   I think it is far better we consume the input from the failed request and then do the ERROR dispatch.

Reply:
All righty.